pytraj/all_actions.py

```diff
@@ -157,6 +158,33 @@ def _assert_mutable(trajiter):
             "This analysis does not support immutable object. Use `pytraj.Trajectory`"
         )
 
+def pair_distance(p1, p2):
+    x1, y1, z1 = p1
+    x2, y2, z2 = p2
+    return np.sqrt((x1 - x2)**2 + (y1 - y2)**2 + (z1 - z2)**2)
+
+@register_pmap
+def closest_atom(top=None, frame=None, point=(0, 0, 0), mask=""):
+    if (top is None) or (len(top.atom_indices(mask)) == 0):
+        raise ValueError("Please pass in a topology file with atoms that match the mask in it")
+    if (frame is None) or (len(frame.coordinates) == 0):
+        raise ValueError("Please pass in a valid frame with coordinates to the function")
+    if (type(point) is not tuple):
+        raise ValueError("Point argument with xyz coordinates must be a tuple")
+    if len(point) != 3:
+        raise ValueError("Please pass in a tuple of length 3, for xyz coordinates")
```

Comment:
I don't think we need to over-check the arguments.
The only check, IMO, that is needed is `len(top.atom_indices(mask)) == 0)`

Adding doc (previous comment) should make thing clearer.